CLI-322: Add '-' as an option char and implemented extensive tests

Claudenw opened the pull request:
Fix for CLI-322

Adds kabab-format options and extensive option name testing.

garydgregory commented:
@Claudenw 
I think you mean "kabab" -> "kebab". Always run `mvn` before you push with the latest from git master.

Claudenw commented:
> @Claudenw I think you mean "kabab" -> "kebab". Always run `mvn` before you push with the latest from git master.

Indeed, I do not know how to spell. :(

I did run 'mvn' before the push and everything worked.  Got "BUILD SUCCESS" result.  Just tried again and same thing.  I am addressing the issues from the build now.

Claudenw commented:
The build fails on 22-ea which fails with a spotbugs execution error,  It seems like spotbugs is not yet supported on 22-ea.

garydgregory commented:
RE 22-ea, that should be fixed by the next commons-parent release, so don't worry about that part.

garydgregory commented on src/main/java/org/apache/commons/cli/OptionValidator.java:
```diff
@@ -23,14 +23,35 @@ Licensed to the Apache Software Foundation (ASF) under one or more
+     * @param c the char to look for.
+     * @return {@code true} if {@code c} was in {@code ary}, {@code false} otherwise. 
+     */
+    private static boolean search(final char[] ary, final char c) {
```
'ary' is a poor parameter name here, almost any else else is better: 'chars', 'array, 'buffer'...

garydgregory reviewed (changes requested):
Hi @Claudenw 
The site needs matching updates to document this feature.

Claudenw commented:
@garydgregory updated as requested

garydgregory commented:
Hi @Claudenw 
Run `mvn` locally before you push to catch these build failures.
